### PR TITLE
fix(ai.training): datatr-364 - training flavor binding and price

### PR DESCRIPTION
**Pull request by jperchoc**
DATATR-364

| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #DATATR-364
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Fix missing binding on flavors and incorrect price for AI Training jobs order funnel

### :flags: Translations

- [x] CMT-4369

### CODE REVIEW

**Review by abullet33 — DISMISSED** [review]

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js [hunk 1]
```diff
@@ -268,6 +277,16 @@ export default class PciTrainingJobsSubmitController {
...
   }
 
+  getFlavorPriceWithResources(flavor) {
+    if (!flavor.catalog) return 0;
+    return this.resourceN * flavor.catalog.priceInUcents * 60;
+  }
+
+  getFlavorPriceTaxWithResources(flavor) {
+    if (!flavor.catalog) return 0;
+    return this.resourceN * flavor.catalog.tax * 60;
```
Comment (sachinrameshn) [inline packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js:287]:
```suggestion
    if (!flavor.catalog) return 0;
    return this.resourceN * flavor.catalog.tax * 60;
```
return flavor.catalog ? this.resourceN * flavor.catalog.tax * 60 : 0;

Comment (sachinrameshn) [inline packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js:287]:
```suggestion
    return flavor.catalog ? this.resourceN * flavor.catalog.tax * 60 : 0;
```

File: packages/manager/modules/pci/src/projects/project/ai/apps/add/components/app-resources/app-resources.controller.js [hunk 2]
```diff
@@ -48,24 +48,55 @@ export default class AppResourcesController {
...
     const replicas = this.appModel.scalingStrategy.autoscaling
       ? this.appModel.scalingStrategy.automatic.replicasMin
       : this.appModel.scalingStrategy.fixed.replicas;
     return resourcePrice * this.appModel.resource.nbResources * replicas;
   }
 
+  getResourcePriceWithResources(flavor) {
+    return (
+      this.getFlavorPrice(flavor).priceInUcents *
+      this.appModel.resource.nbResources *
```
[the remark this answers is not in the document]
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/ai/apps/add/components/app-resources/app-resources.controller.js:82]: Done: https://github.com/ovh/manager/pull/10025/commits/bd8a72ae5c10801a5c985f9c69df4cff8744cf79

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js (hunk 1 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.controller.js]: Changed: https://github.com/ovh/manager/pull/10025/commits/bd8a72ae5c10801a5c985f9c69df4cff8744cf79

**Review by abullet33 — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Review by BrunoMarquesOVH — APPROVED** [review]

**Review by ghyenne — APPROVED** [review]

**Pull request merged**